src/mds/Server.cc

```diff
@@ -1212,9 +1223,20 @@ void Server::find_idle_sessions()
 	break;
       }
       to_evict.push_back(session);
+      if (any_osd_laggy) {
+	laggy_sessions.insert(session);
+      }
     }
   }
 
+  // don't kick client if osds are laggy
+  if(get_laggy_sessions()) {
```

Comment:
add mds config to turn off this behavior (skipping eviction if osds are laggy)

Reply:
something like `ceph fs set <fs> evict_clients_on_laggy_osds false`?